app/app/Http/Requests/Auth/LoginRequest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace App\Http\Requests\Auth;
+
+use Illuminate\Auth\Events\Lockout;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\RateLimiter;
+use Illuminate\Support\Str;
+use Illuminate\Validation\ValidationException;
+
+class LoginRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'email' => ['required', 'string', 'email'],
+            'password' => ['required', 'string'],
+        ];
+    }
+
+    /**
+     * Attempt to authenticate the request's credentials.
+     *
+     * @return void
+     *
+     * @throws \Illuminate\Validation\ValidationException
+     */
+    public function authenticate()
```

Comment:
@MaxRazen please take into account that placing logic in Request is a violation of SOLID principles. Please use Service class to store application logic.

Reply:
@vsobchuk good catch, agree
Do I also need to re-struct the app to DDD?

Reply:
@vsobchuk https://github.com/MaxRazen/cookbook-demo/pull/2/files#diff-7b1304d080ba2bd87abc8e5cba2aa2f8aaf82e604b87bf35d810135d91cac7a0